include/roboteam_ai/world/FieldComputations.h

```diff
@@ -23,25 +21,214 @@ namespace rtt::ai {
-
-    static Line getGoalSides(const Field &field, bool ourGoal);
-    static double getDistanceToGoal(const Field &field, bool ourGoal, const Vector2 &point);
-    static Vector2 getPenaltyPoint(const Field &field, bool ourGoal);
-    static std::shared_ptr<Vector2> lineIntersectionWithDefenceArea(const Field &field, bool ourGoal, const Vector2 &lineStart, const Vector2 &lineEnd, double margin);
-    static double getTotalGoalAngle(const Field &field, bool ourGoal, const Vector2 &point);
-    static Polygon getDefenseArea(const Field &field, bool ourDefenseArea = true, double margin = 0.0, bool includeOutSideField = true);
-    static Polygon getGoalArea(const Field &field, bool ourGoal = true, double margin = 0.0, bool hasBackMargin = false);
-    static Polygon getFieldEdge(const Field &field, double margin = 0.0);
+    private:
+        static constexpr double NEGLIGIBLE_LENGTH = 0.000001; // If a line length is below or equal to this threshold then is it neglected during determining the blockades.
+
+    public:
+        /**
+         * Determines whether a given point is in our/their defence area
+         * @param field The field class which is used to determine the position of the defence areas.
+         * @param point The point for which it is checked whether it is in our/their defence area.
+         * @param isOurDefenceArea True if it is checked whether the point is in our defence area, false if it is checked whether the point is in the opponents defence area (if
+         * this parameter is not set then it checks whether the point is in our defence area).
+         * @param margin The outwards margin with which the defence area will be expanded/shrinked for determining whether the point is in our defence area. A positive value means
+         * that the rectangular defence area will be expanded in both y directions and towards the center of the field, a negative value means that the rectangular defence area
+         * will be shrinked in both y directions and away from the goal side of that defence area (if this parameter is not set then the defence area will not be expanded/shrinked)
+         * @param includeOutsideField If set to true then the defence area will also be expanded/shrinked at the goal side (boundary side) of the field, if set to false then it
+         * will not be expanded/shrinked at the goal side (boundary side) of the field (if this parameter is not set then the defence area will not be expanded/shrinked at the goal
+         * side (boundary side) of the field).
+         * @return True if the point is in the given defence area, false if the point is not in the given defence area.
+         */
+        static bool pointIsInDefenceArea(const Field &field, const Vector2& point, bool isOurDefenceArea = true, double margin = 0.0, bool includeOutsideField = false);
```

Comment:
it is so obvious what this function does. if you rename margin to "outsidemargin" you are basically there.

Reply:
Well according to the rules all public functions need documentation where all parameters and return statements need to be documented. And I am consistent in this principle. I agree that it is indeed quite clear what it does and therefore I keep it short. The outside margin is indeed something that should be mentioned, because it might confuse developers.

Reply:
Which rules?

Reply:
Issue with anything that's not the C++ standard is that there are no rules

Reply:
> Which rules?

According to our unwritten rules

Reply:
I actually think we should write them down :eyes:

Reply:
I indeed agree, but it does not have the highest priority level compared to other adaptive maintenance tasks. So probably it will be postponed, unless someone else wants to write them down.